app/uk/gov/hmrc/tai/connectors/DesConnector.scala

```diff
@@ -50,44 +49,49 @@ class DesConnector @Inject()(
 
   def daPtaOriginatorId = config.daPtaOriginatorId
 
-  def desPathUrl(nino: Nino, path: String) = s"${config.baseURL}/pay-as-you-earn/individuals/$nino/$path"
+  private def desPathUrl(nino: Nino, path: String) = s"${config.baseURL}/pay-as-you-earn/individuals/$nino/$path"
 
-  def commonHeaderValues =
+  private def commonHeaderValues(implicit hc: HeaderCarrier) =
     Seq(
-      "Environment"   -> config.environment,
-      "Authorization" -> config.authorization,
-      "Content-Type"  -> TaiConstants.contentType)
-
-  def header: HeaderCarrier = HeaderCarrier(extraHeaders = commonHeaderValues)
-
-  def headerForUpdate(version: Int, originatorId: String): HeaderCarrier =
-    HeaderCarrier(extraHeaders = commonHeaderValues ++ Seq("Originator-Id" -> originatorId, "ETag" -> version.toString))
+      "Environment"          -> config.environment,
+      "Authorization"        -> config.authorization,
+      "Content-Type"         -> TaiConstants.contentType,
+      HeaderNames.xSessionId -> hc.sessionId.fold("-")(_.value),
+      HeaderNames.xRequestId -> hc.requestId.fold("-")(_.value),
+      "CorrelationId"        -> UUID.randomUUID().toString
+    )
+
+  private def header(implicit hc: HeaderCarrier): HeaderCarrier = HeaderCarrier(extraHeaders = commonHeaderValues)
```

Comment:
rename this to something like createDesHeader

Reply:
done